From 6012c6e4087719f3bfe051d9658e60b061c57199 Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <yamada.masahiro@socionext.com>
Date: Tue, 22 Mar 2016 21:13:32 +0900
Subject: [PATCH 05/16] arm64: allow to enable I-cache when switching to EL2/1

Signed-off-by: Masahiro Yamada <yamada.masahiro@socionext.com>
---
 arch/arm/include/asm/macro.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/include/asm/macro.h b/arch/arm/include/asm/macro.h
index bb33b4b..4a2b724 100644
--- a/arch/arm/include/asm/macro.h
+++ b/arch/arm/include/asm/macro.h
@@ -169,6 +169,9 @@ lr	.req	x30
 			SCTLR_EL2_WXN_DIS | SCTLR_EL2_ICACHE_DIS |\
 			SCTLR_EL2_SA_DIS | SCTLR_EL2_DCACHE_DIS |\
 			SCTLR_EL2_ALIGN_DIS | SCTLR_EL2_MMU_DIS)
+#ifndef CONFIG_SYS_ICACHE_OFF
+	orr	\tmp, \tmp, #(1 << 12)
+#endif
 	msr	sctlr_el2, \tmp
 
 	mov	\tmp, sp
@@ -276,6 +279,9 @@ lr	.req	x30
 			SCTLR_EL1_CP15BEN_DIS | SCTLR_EL1_SA0_DIS |\
 			SCTLR_EL1_SA_DIS | SCTLR_EL1_DCACHE_DIS |\
 			SCTLR_EL1_ALIGN_DIS | SCTLR_EL1_MMU_DIS)
+#ifndef CONFIG_SYS_ICACHE_OFF
+	orr	\tmp, \tmp, #(1 << 12)
+#endif
 	msr	sctlr_el1, \tmp
 
 	mov	\tmp, sp
-- 
2.7.4

